### ISSUE TITLE
mongodb externalAccess doesn't work as expected

### ISSUE THREAD

**Issue by richert-addisca** [bitnami/charts#7961]
**Which chart**:
mongodb 10.28.5

**Describe the bug**
When installing the chart with the following options:
```
rbac:
  create: true
  role:
    rules:
      - apiGroups:
          - ""
        resources:
          - pods
        verbs:
          - get
          - list
          - watch
          - update
externalAccess:
  enabled: true
  autoDiscovery:
    enabled: true
```
It will create services for each pod with a separate IP as expected, and you can connect to each of them individually as you would expect, but if you want to access the ReplicaSet, aka. all of them together you get the following error:
```
MongoNetworkError: getaddrinfo ENOTFOUND auth-mongodb-0.auth-mongodb-headless.NAMESPACE.svc.cluster.local
```
The above error is thrown by mongosh, if you try to connect via mongoose in NodeJS you get the more elaborate error:
```
MongooseServerSelectionError: getaddrinfo ENOTFOUND auth-mongodb-0.auth-mongodb-headless.NAMESPACE.svc.cluster.local
    at NativeConnection.Connection.openUri (C:\Users\USER\APP\node_modules\mongoose\lib\connection.js:797:32)
    at Mongoose.createConnection (C:\Users\USER\APP\node_modules\mongoose\lib\index.js:274:10)
    at Object.<anonymous> (C:\Users\USER\APP\connections\db_user.js:20:23)
    at Module._compile (internal/modules/cjs/loader.js:1072:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1101:10)
    at Module.load (internal/modules/cjs/loader.js:937:32)
    at Function.Module._load (internal/modules/cjs/loader.js:778:12)
    at Module.require (internal/modules/cjs/loader.js:961:19)
    at require (internal/modules/cjs/helpers.js:92:18)
    at Object.<anonymous> (C:\Users\USER\APP\routes\heartbeat.js:3:18) {
  reason: TopologyDescription {
    type: 'ReplicaSetNoPrimary',
    servers: Map(4) {
      'auth-mongodb-0.auth-mongodb-headless.NAMESPACE.svc.cluster.local:27017' => [ServerDescription],
      'auth-mongodb-1.auth-mongodb-headless.NAMESPACE.svc.cluster.local:27017' => [ServerDescription],
      'auth-mongodb-2.auth-mongodb-headless.NAMESPACE.svc.cluster.local:27017' => [ServerDescription],
      'auth-mongodb-arbiter-0.auth-mongodb-arbiter-headless.NAMESPACE.svc.cluster.local:27017' => [ServerDescription]
    },
    stale: false,
    compatible: true,
    heartbeatFrequencyMS: 10000,
    localThresholdMS: 15,
    setName: 'rs0',
    maxSetVersion: 4,
    maxElectionId: new ObjectId("7fffffff0000000000000002"),
    commonWireVersion: 9,
    logicalSessionTimeoutMinutes: undefined
  }
}

```

**To Reproduce**

1. Install the chart with the aforementioned values
2. Try to connect to the replica set

**Expected behavior**
A working connection

**Version of Helm and Kubernetes**:

- Output of `helm version`:

```
version.BuildInfo{Version:"v3.6.3", GitCommit:"d506314abfb5d21419df8c7e7e68012379db2354", GitTreeState:"clean", GoVersion:"go1.16.5"}
```

- Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.8", GitCommit:"5575935422cc1cf5169dfc8847cb587aa47bac5a", GitTreeState:"clean", BuildDate:"2021-06-16T13:00:45Z", GoVersion:"go1.15.13", Compiler:"gc", Platform:"windows/amd64"}
Server Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.10", GitCommit:"8152330a2b6ca3621196e62966ef761b8f5a61bb", GitTreeState:"archive", BuildDate:"2021-08-19T11:59:03Z", GoVersion:"go1.15.15", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by miguelaeh** [comment]
Hi @richert-addisca ,
I don't fully understand how are you connecting to the service. Could you please be more specific in which endpoint did you use and if it is from inside or outside of the K8s cluster?
Note that from outside the cluster you need to use the LoadBalancer IPs, you can only use the service name from inside the cluster.

**Comment by richert-addisca** [comment]
Hi @miguelaeh,
as mentioned above I am deploying the bitnami/mongodb helm skript with the above values.

I get these 3 services of type LoadBalancer:

- database-mongodb-0-external
- database-mongodb-1-external
- database-mongodb-2-external

all of which have their own public IP.

As mentioned above I can connect to each of them individually, but can not connect to the replicaSet.

I am trying to connect from outside the cluster with the public IP adresses, but trying to connect to the cluster results in an error where, for some reason, mongoose, as well as mongosh try to connect to the internal adresses which, of course, fails.

**Comment by miguelaeh** [comment]
Hi @richert-addisca ,
Two points here then:

1. About getting external access to the full replicaset, I am afraid it is not supported right now on this chart due to its own nature, but since you have access to the 3 external IPs, you can always configure by yourself a load balancer grouping them, so have a single entry point.
2.
>for some reason, mongoose, as well as mongosh try to connect to the internal adresses which, of course, fails.

About this, it seems you forgot to perform the upgrade setting the LB IPs following the instructions on the NOTES.txt that are printed when you install the chart. Could you check it?

**Comment by richert-addisca** [comment]
OK, a strange thing happened. I just installed a new release with helm and the exact same settings but in a new namespace. And now it works as intended.

I did install a release with helm before without external access, and then i uninstalled it and installed a new one with the above settings (not upgraded).

So it seems that for some reason a fresh install does not remove all the previous settings.

Is there a possibility for a clean up to remove potential old settings.

I mean I don't even understand how this can happen, I even deleted all the persistant volumes that where used by mongo before reinstall.
